### PR TITLE
fix: Added keyboard functionality for esc and Ctrl + a

### LINKED ISSUE #8247: Pressing ⎋ on indexes should deselect all records
# Current behavior

Nothing happens when pressing `⎋ (escape)` on an index:

https://github.com/user-attachments/assets/c3a90858-2370-49ad-a247-7732f26c3061

# Desired Behavior

We should deselect all records when pressing escape.

### PR DESCRIPTION
**Pull request by harshit078**
## Description

- This PR fixes 
#8247 
#8206 
- Adds keyboard functionality for esc (deselect all records) and Ctrl + a (select records)

## Changes

https://github.com/user-attachments/assets/7b2718d6-dbe5-4cd8-8ead-b6baed1fc357

### CODE REVIEW

**Comment by Devessier** [comment]
Thank you a lot for your contribution, @harshit078. We tried to find a more global solution to the problem.

My implementation is still in progress. Things to finish:

- [ ] Use scoped hotkeys for the board as well

**Comment by Devessier** [comment]
Hello @harshit078, would you be able to continue working on this feature? I will try to do it myself if not. Thank you so much!

**Comment by Devessier** [comment]
It seems that the `TableHotkeyScope.Table` and `TableHotkeyScope.Table` scopes are no longer restored when the selection ends. As a consequence, the listeners I wrote are no longer working.

@lucasbordeau or @charlesBochet, do you know what I should do instead? Thanks for any help.

I will put this PR on standby for now.

File: packages/twenty-front/src/modules/object-record/record-board/components/RecordBoard.tsx [hunk 1]
```diff
@@ -61,6 +62,7 @@ export const RecordBoard = () => {
     columnIdsState,
     columnsFamilySelector,
     recordIdsByColumnIdFamilyState,
+    allRecordIdsSelector,
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-board/components/RecordBoard.tsx:65]: we could refactor this into record-index to mutualize this behavior between table and board but I don't think this is the topic of this PR

File: packages/twenty-front/src/modules/object-record/record-board/components/RecordBoard.tsx [hunk 2]
```diff
@@ -80,7 +82,28 @@ export const RecordBoard = () => {
...
+          .getValue();
+
+        for (const recordId of allRecordIds) {
+          setRecordAsSelected(recordId, true);
+        }
+      },
+    [allRecordIdsSelector, setRecordAsSelected],
+  );
+
+  useScopedHotkeys('ctrl+a,meta+a', selectAll, TableHotkeyScope.Table);
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-board/components/RecordBoard.tsx:99]: TableHotkeyScope.Table unlikely to work on the board

File: packages/twenty-front/src/modules/object-record/record-table/components/RecordTableWithWrappers.tsx [hunk 3]
```diff
@@ -40,6 +46,23 @@ export const RecordTableWithWrappers = ({
   recordTableId,
   viewBarId,
 }: RecordTableWithWrappersProps) => {
+  const { resetTableRowSelection, selectAllRows } = useRecordTable({
+    recordTableId,
+  });
+
+  useScopedHotkeys(
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-table/components/RecordTableWithWrappers.tsx:53]: this should be moved to ActionBar

**Comment by Devessier** [comment]
It's ready for a new review, @charlesBochet.

**Review by charlesBochet — APPROVED** [review]

**Comment by FelixMalfait** [comment]
@Devessier fyi it seems that cmd+a doesn't have the same behavior as clicking on the top checkbox. Clicking on the checkbox selects all records on the backend (e.g. 1M+ records), while cmd+a selects the visible records (e.g. 120 records if you've loaded 2 pages on the frontend). Could you please adjust that?

**Pull request merged**